internal/config/types.go

```diff
@@ -52,12 +58,14 @@ type TLS struct {
 // - values in environment variables
 // Array/Slice types are not supported within this data-structure.
 type Config struct {
-	Log                     Log           `yaml:"log" json:"log" env-prefix:"KOKO_LOG_"`
-	Admin                   AdminServer   `yaml:"admin_server" json:"admin_server" env-prefix:"KOKO_ADMIN_SERVER_"`
-	Control                 ControlServer `yaml:"control_server" json:"control_server" env-prefix:"KOKO_CONTROL_SERVER_"`
-	Database                Database      `yaml:"database" json:"database" env-prefix:"KOKO_DATABASE_"`
-	Metrics                 Metrics       `yaml:"metrics" json:"metrics" env-prefix:"KOKO_METRICS_"`
-	DisableAnonymousReports bool          `yaml:"disable_anonymous_reports" json:"disable_anonymous_reports" env-prefix:"KOKO_DISABLE_ANONYMOUS_REPORTS"` //nolint:lll
+	Log      Log           `yaml:"log" json:"log" env-prefix:"KOKO_LOG_"`
+	Admin    AdminServer   `yaml:"admin_server" json:"admin_server" env-prefix:"KOKO_ADMIN_SERVER_"`
+	Control  ControlServer `yaml:"control_server" json:"control_server" env-prefix:"KOKO_CONTROL_SERVER_"`
+	Database Database      `yaml:"database" json:"database" env-prefix:"KOKO_DATABASE_"`
+	// Deprecated
+	Metrics                 Metrics           `yaml:"metrics" json:"metrics" env-prefix:"KOKO_METRICS_"`
```

Comment:
Can we actually keep this around, deprecate the fields inside that are unused, and more Prometheus in there? I say that because there may be a global-scoped metric config that may impact all integrations in the future, and that'll at least set us up for that (just in case).

The struct would look something like:
```go
type Metrics struct {
    // Deprecated: This is kept around to enable the Datadog metrics integration
    // to continue to work, until we're ready to switch over to Prometheus.
	ClientType string            `yaml:"client_type" json:"client_type" env:"CLIENT_TYPE" env-default:"noop"`
    
    Prometheus PrometheusMetrics `yaml:"log" json:"log" env-prefix:"PROMETHEUS_"`
}

// PrometheusMetrics defines the configuration for setting up the Prometheus
// metrics exporter, which exposes an HTTP server returning metrics in the
// Prometheus exposition format under the `/metrics` HTTP path. 
type PrometheusMetrics struct {
	Enable  bool   `yaml:"enable" json:"enable" env:"ENABLE"`
	Address string `yaml:"address" json:"address" env:"ADDRESS" env-default:":9090"`
}
```